src/general_tally.cpp

```diff
@@ -0,0 +1,243 @@
+#include <tallies/general_tally.hpp>
+#include <utils/error.hpp>
+#include <utils/output.hpp>
+
+#include <boost/container/static_vector.hpp>
+using StaticVector4 = boost::container::static_vector<size_t, 4>;
+
+GeneralTally::GeneralTally(std::shared_ptr<PositionFilter> position_filter,
+                           std::shared_ptr<EnergyFilter> energy_in,
+                           Quantity quantity, Estimator estimator,
+                           std::string name_)
+    : ITally(quantity, estimator, name_),
+      position_filter_(position_filter),
+      energy_in_(energy_in) {
+  // at least one filter must exists.
+  if (position_filter_ == nullptr && energy_in_ == nullptr) {
+    fatal_error("for the " + name_ + " tally, no filter is provided.");
+  }
```

Comment:
We shouldn't require at least one filter. We want to be able to integrate over all space and energy for certain things.